### PR TITLE
cmd/lxd-to-incus: Uncompress snapshot files in target database

### LINKED ISSUE #66: Disable raft snapshot compression
I've run a few tests and benchmarks around cowsql database raft snapshot compression using lz4, and the conclusion seems to be that they are not worth, since they save relatively little space at the price of noticeable CPU overhead.

The database of an incus deployment with 1000 instances and associated metadata (profiles, networks, snapshots, images, volumes, etc) has an uncompressed size of ~4M and a compressed size of ~3M, so lz4 doesn't really move the needle there. A deployment of 50k instances has a database size of ~181M, and a compressed size of ~150M: the storage/memory gain is modest, while the CPU overhead of running compression can be noticeable when hitting the API when the snapshot is in progress.

A much more effective measure seems to be to regularly `VACUUM` the database, which can reduce the size of a database dramatically (even 10x for databases that have been running for a long time and have a lot of fragmentation). I'm planning to run VACUUM transparently in cowsql.

If there's consensus around this, I'd like to remove compression from both incus and cowsql and figure the best option for upgrades (it'd basically just a matter of uncompressing the existing snapshots at upgrade time and deciding where to put that logic).

This should also simplify the on-disk format and make it easier to improve the command line tooling that we have around it.

### PR DESCRIPTION
**Pull request by freeekanayaka**
Fixes #66

### CODE REVIEW

File: cmd/lxd-to-incus/db.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
+package main
+
+import (
+	"fmt"
+	"github.com/lxc/incus/shared"
+	"github.com/pierrec/lz4/v4"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
```
Comment (stgraber) [inline cmd/lxd-to-incus/db.go:10]: We usually group those like:
```
"fmt"
"io"
"io/ioutil"
"os"
"path/filepath"

"github.com/pierrec/lz4/v4"

"github.com/lxc/incus/shared"
```

Basically: stdlib, external, local

File: cmd/lxd-to-incus/db.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+		if err != nil {
+			return fmt.Errorf("Failed to uncompress snapshot %q: %w", filename, err)
+		}
+	}
+
+	return nil
+}
+
+// Uncompress the given file, preserving its mode and ownership.
+func uncompress(zfilename string) error {
```
Comment (stgraber) [inline cmd/lxd-to-incus/db.go:54]: Maybe `lz4Uncompress` so it's clear that it does just lz4?

File: cmd/lxd-to-incus/db.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+	if err != nil {
+		return fmt.Errorf("Failed to open file %q: %w", zfilename, err)
+	}
+
+	zinfo, err := zfile.Stat()
+	if err != nil {
+		return fmt.Errorf("Fialed to get file info for %q: %w", zfilename, err)
+	}
+
+	mode := zinfo.Mode() // use the same mode for the output file
```
Comment (stgraber) [inline cmd/lxd-to-incus/db.go:67]: Tiny nit but we usually prefer the comments to be on their own line :)

File: cmd/lxd-to-incus/main.go [hunk 4]
```diff
@@ -395,6 +395,11 @@ Instances will come back online once the migration is complete.
 		return fmt.Errorf("Failed to move %q to %q: %w", sourcePaths.Daemon, targetPaths.Daemon, err)
 	}
 
+	err = migrateDatabase(filepath.Join(targetPaths.Daemon, "database"))
```
Comment (stgraber) [inline cmd/lxd-to-incus/main.go:398]: I'd add another status update here for the user.
```
fmt.Println("=> Migrating database")
```

**Review by stgraber — CHANGES_REQUESTED** [review]

**Comment by stgraber** [comment]
@freeekanayaka so this will uncompress the database on migration, but I guess we need another change to have Incus not set it up to be compressed in the first place, right?

Otherwise as soon as we start Incus, it will just go back to being compressed again, no?

**Comment by freeekanayaka** [comment]
> @freeekanayaka so this will uncompress the database on migration, but I guess we need another change to have Incus not set it up to be compressed in the first place, right?
> 
> Otherwise as soon as we start Incus, it will just go back to being compressed again, no?

Not really, as soon as this lands I'd disable/remove compressing entirely from cowsql.

File: cmd/lxd-to-incus/db.go (hunk 2 above)
Comment (freeekanayaka) [reply cmd/lxd-to-incus/db.go]: Fixed.

File: cmd/lxd-to-incus/db.go (hunk 3 above)
Comment (freeekanayaka) [reply cmd/lxd-to-incus/db.go]: Fixed.

**Review by stgraber — APPROVED** [review]

**Pull request merged**